### PR TITLE
Roe 2413 bo other url params

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2413

### Change description

Updated:
- Index
- routes
- navigation
- bo other controller post
- bo other unit tests get and post
- 
### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: views/beneficial-owner-type.html
Comment (markpit) [inline views/beneficial-owner-type.html:77]: we should have the one feature flag check and override the feature flag "off" values in that block.
so - {% set beneficialOwnerOtherUrl = OE_CONFIGS.BENEFICIAL_OWNER_OTHER_URL %} 
should go above the feature flag block and in the block override it with
{% set beneficialOwnerOtherUrl = beneficialOwnerOtherUrlWithParams %}

File: src/routes/index.ts
Comment (markpit) [inline src/routes/index.ts:219]: we should also update the 
config.BENEFICIAL_OWNER_OTHER_URL + config.REMOVE + config.ID
route as well to make sure the remove button works as well as the change button

Comment (markpit) [reply src/routes/index.ts:221]: hang on, i think it is already there

Comment (markpit) [reply src/routes/index.ts:221]: yes can see it, ignore this .

File: views/beneficial-owner-type.html
Comment (mattch1) [reply views/beneficial-owner-type.html:70]: Consolidated this.

File: src/routes/index.ts
Comment (markpit) [inline src/routes/index.ts:245]: can you group this remove route with the non-params one and separate from other groups with blank line. eg, line 244-245 needs blank line separator so we can see the routes more easily

File: src/routes/index.ts
Comment (mattch1) [reply src/routes/index.ts:221]: Don't know how to get this to work, when remove is clicked the ids are undefined not sure what is wiping them out.

File: src/controllers/beneficial.owner.type.controller.ts
Comment (dlloyd1-ch) [inline src/controllers/beneficial.owner.type.controller.ts:87]: We already check if the flag is active in this function. Code can be refactored to check the flag for all urls. Otherwise we'll end up checking for the flag for each url.
e.g.
```
 if (isActiveFeature(config.FEATURE_FLAG_ENABLE_REDIS_REMOVAL)){
    if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.individual) {
      return getUrlWithParamsToPath(config.BENEFICIAL_OWNER_INDIVIDUAL_WITH_PARAMS_URL, req);
    } else if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.otherLegal) {
      return getUrlWithParamsToPath(config.BENEFICIAL_OWNER_OTHER_WITH_PARAMS_URL, req);
    } ...
  } else {
    if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.individual) {
      return config.BENEFICIAL_OWNER_INDIVIDUAL_URL;
    } else if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.otherLegal) {
      return config.BENEFICIAL_OWNER_OTHER_URL;
    } ...
}
```

Comment (mattch1) [reply src/controllers/beneficial.owner.type.controller.ts:87]: The trouble is we then end up checking this twice:
if (beneficialOwnerTypeChoices === BeneficialOwnerTypeChoice.individual)
so I'm not sure what we gain.

Comment (dlloyd1-ch) [reply src/controllers/beneficial.owner.type.controller.ts:87]: By checking the flag only once we have all the feature flagged code in one place, which will make it easier when we come to remove the flag in the future. Otherwise we'll eventually have 5 flag blocks of code to remove - one for each URL in this function.

Comment (mattch1) [reply src/controllers/beneficial.owner.type.controller.ts]: Just realised this before I read your reply have updated this into blocks as shown.

File: src/routes/index.ts
Comment (mattch1) [reply src/routes/index.ts:245]: Updated this to be grouped.

**Comment by mwestacott** [comment]
Not sure if these PRs are being merged into the `feature/roe-1989-add-ids-into-registration-urls` branch, as that's about to be merged into `main`.

File: src/controllers/beneficial.owner.other.controller.ts
Comment (mwestacott) [inline src/controllers/beneficial.owner.other.controller.ts:40]: Wondering why this function is named `nextOrLastPage`? I did a similar thing in my [PR](https://github.com/companieshouse/overseas-entities-web/pull/1145) and named it `getNextPageUrl`.

**Comment by dlloyd1-ch** [comment]
> Not sure if these PRs are being merged into the `feature/roe-1989-add-ids-into-registration-urls` branch, as that's about to be merged into `main`.

These PRs should not be merged into the feature/roe-1989... Once that's been merged, the PRs should automatically repoint to 'main'

File: src/controllers/beneficial.owner.other.controller.ts
Comment (mattch1) [reply src/controllers/beneficial.owner.other.controller.ts:40]: It is used by the get methods and the arguments there are for the previous page, so covered both with the name.

Comment (mwestacott) [reply src/controllers/beneficial.owner.other.controller.ts:40]: I think it's simply the next page URL that the app is going to redirect to. But up to you.

Comment (mattch1) [reply src/controllers/beneficial.owner.other.controller.ts:38]: Keep as is, if its a backlink next page could be misleading.

File: src/controllers/beneficial.owner.other.controller.ts
Comment (markpit) [inline src/controllers/beneficial.owner.other.controller.ts:39]: I agree with Martin about the function name, perhaps getBeneficialOwnerTypeUrl() would be better? it would avoid the not knowing if it was in fact a next page or previous page.

Comment (mattch1) [reply src/controllers/beneficial.owner.other.controller.ts:39]: Updated the name as suggested.

**Review by mwestacott — APPROVED** [review]
Looks good to me but get approval from @markpit too before merging please.

**Review by markpit — APPROVED** [review]

**Pull request merged**
